client/types.ts

```diff
@@ -6,6 +6,13 @@ import { NonUndefined } from 'utility-types';
 // Web stuff
 export type URL = string;
 export type Scheme = 'http' | 'https';
+export type JSONSerializable =
+	| null
+	| boolean
+	| number
+	| string
+	| JSONSerializable[]
+	| { [ prop: string ]: JSONSerializable };
```

Comment:
small nitpick that doesn't matter much, but I'd personally put this down below under `Language stuff` since it's more part-and-parcel to JS the language